source/Crew.cpp

```diff
@@ -0,0 +1,242 @@
+/* Crew.cpp
+Copyright (c) 2019 by Luke Arndt
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+*/
+
+#include "Crew.h"
+#include "GameData.h"
+
+using namespace std;
+
+// Load definition of a crew member
+void Crew::Load(const DataNode &node)
+{
+	// Set the id of this crew member so we know that we have loaded it.
+	if(node.Size() >= 2)
+		id = node.Token(1);
+	
+	for(const DataNode &child : node)
+	{
+		if(child.Size() >= 2)
+		{
+			// The crew members' display name (plural, Title Case)
+			if(child.Token(0) == "name")
+				name = child.Token(1);
+			// Each valid ship has at least this many of the crew member
+			else if(child.Token(0) == "minimum per ship")
+				minimumPerShip = max((int)child.Value(1), 0);
+			// The number of credits paid daily while parked (minimum 0)
+			else if(child.Token(0) == "parked salary")
+				parkedSalary = max((int)child.Value(1), 0);
+			// Every nth crew member on the ship will be this crew member
+			else if(child.Token(0) == "population per member")
+				populationPerMember = max((int)child.Value(1), 0);
+			// The number of credits paid daily (minimum 0)
+			else if(child.Token(0) == "salary")
+				salary = max((int)child.Value(1), 0);
+			else
+				child.PrintTrace("Skipping unrecognized attribute:");
+		}
+		// If true, the crew member will not appear on escorts
+		else if(child.Token(0) == "avoids escorts")
+			avoidsEscorts = true;
+		// If true, the crew member will not appear on the flagship
+		else if(child.Token(0) == "avoids flagship")
+			avoidsFlagship = true;
+		else
+			child.PrintTrace("Skipping incomplete attribute:");
+	}
+}
+
+
+
+int64_t Crew::CalculateSalaries(
+	const vector<shared_ptr<Ship>> &ships,
+	const Ship * &flagship,
+	const bool includeExtras
+)
+{
+	int64_t totalSalaries = 0;
+	
+	for(const shared_ptr<Ship> &ship : ships)
+	{
+		totalSalaries += Crew::SalariesForShip(
+			ship,
+			ship.get() == flagship,
```

Comment:
It's a bit micro-optimization, but you could set a bool flag when you find the flagship so that you can skip this pointer dereference + comparison check afterwards.

Reply:
Yes, that would help performance, good idea.